js-extensions/src/ClassMetadataStore.js

```diff
@@ -8,13 +8,20 @@ export class ClassMetadataStore {
          * Type info cache
          */
         this.classMetadata = {};
-        
+
         /**
          * Fetch function for the classMetadata
          */
         this.classMetadataProvider = classMetadataProvider;
+
+        /**
+         * Onload callbacks cache. Used to ensure we don't
+         * issue multiple in-parallel requests for the same
+         * class metadata.
+         */
+        this.classMetadataOnloadCallbacks = {};
```

Comment:
@tfennelly agree, unless it's excessively slow, we should not make this change only to revert it later (soon) when we fix it at a lower level